src/packages/sdk/src/engine-commands/getConfig.ts

```diff
@@ -0,0 +1,156 @@
+import Debug from '@prisma/debug'
+import { NApiEngineTypes } from '@prisma/engine-core'
+import { EngineTypes } from '@prisma/fetch-engine'
+import { DataSource, GeneratorConfig } from '@prisma/generator-helper'
+import chalk from 'chalk'
+import execa from 'execa'
+import fs from 'fs'
+import tmpWrite from 'temp-write'
+import { promisify } from 'util'
+import { resolveBinary } from '../resolveBinary'
+
+const debug = Debug('prisma:getConfig')
+
+const unlink = promisify(fs.unlink)
+
+const MAX_BUFFER = 1_000_000_000
+
+export interface ConfigMetaFormat {
+  datasources: DataSource[]
+  generators: GeneratorConfig[]
+  warnings: string[]
+}
+
+export type GetConfigOptions = {
+  datamodel: string
+  cwd?: string
+  prismaPath?: string
+  datamodelPath?: string
+  retry?: number
+  ignoreEnvVarErrors?: boolean
+}
+
+export async function getConfig({
+  datamodel,
+  cwd = process.cwd(),
+  prismaPath: queryEnginePath,
+  datamodelPath,
+  ignoreEnvVarErrors,
+}: GetConfigOptions): Promise<ConfigMetaFormat> {
+  const useNapi = process.env.PRISMA_FORCE_NAPI === 'true'
+  if (useNapi) {
+    queryEnginePath = await resolveBinary(
+      EngineTypes.libqueryEngineNapi,
+      queryEnginePath,
+    )
+  } else {
+    queryEnginePath = await resolveBinary(
+      EngineTypes.queryEngine,
+      queryEnginePath,
+    )
+  }
+  let data: ConfigMetaFormat | undefined
+  debug(`Using ${useNapi ? 'N-API ' : ''}Query Engine at: ${queryEnginePath}`)
+  if (useNapi) {
+    try {
+      const NApiQueryEngine = require(queryEnginePath) as NApiEngineTypes.NAPI
+      data = await NApiQueryEngine.getConfig({
+        datamodel: datamodel,
+        datasourceOverrides: {},
+        ignoreEnvVarErrors: ignoreEnvVarErrors ?? false,
+      })
+    } catch (e) {
+      let error
+      try {
+        error = JSON.parse(e.message)
+      } catch {
+        throw e
+      }
+      let message: string
+      if (error.error_code === 'P1012') {
+        message =
+          chalk.redBright(`Schema Parsing ${error.error_code}\n\n`) +
+          error.message +
+          '\n'
+      } else {
+        message = chalk.redBright(`${error.error_code}\n\n`) + error
+      }
+      throw new GetConfigError(message)
+    }
+  } else {
+    try {
+      let tempDatamodelPath: string | undefined = datamodelPath
+      if (!tempDatamodelPath) {
+        try {
+          tempDatamodelPath = await tmpWrite(datamodel!)
+        } catch (err) {
+          throw new GetConfigError('Unable to write temp data model path')
+        }
+      }
+      const engineArgs = []
+
+      const args = ignoreEnvVarErrors ? ['--ignoreEnvVarErrors'] : []
+
+      const result = await execa(
```

Comment:
What is this case about?

Reply:
This how getConfig is currently called using a Binary